tests/snuba/api/endpoints/test_organization_events_stats_mep.py

```diff
@@ -1150,3 +1149,64 @@ def test_top_events_with_transaction_on_demand_and_no_environment(self):
             group, agg, row1, row2 = group_count
             row_data = response.data[group][agg]["data"][:2]
             assert [attrs for time, attrs in row_data] == [[{"count": row1}], [{"count": row2}]]
+
+    def test_timeseries_on_demand_with_multiple_percentiles(self):
+        field = "p75(measurements.fcp)"
+        field_two = "p75(measurements.lcp)"
+        query = "transaction.duration:>=100"
+        spec = OnDemandMetricSpec(field=field, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY)
+        spec_two = OnDemandMetricSpec(
+            field=field_two, query=query, spec_type=MetricSpecType.DYNAMIC_QUERY
+        )
+
+        assert (
+            spec._query_str_for_hash
+            == "event.measurements.fcp.value;{'name': 'event.duration', 'op': 'gte', 'value': 100.0}"
+        )
+        assert (
+            spec_two._query_str_for_hash
+            == "event.measurements.lcp.value;{'name': 'event.duration', 'op': 'gte', 'value': 100.0}"
+        )
+
+        # Include transaction duration in indexer since it's in the query.
+        indexer.record(UseCaseID.TRANSACTIONS, self.organization.id, "transaction.duration")
```

Comment:
Why do we not need it in some other test cases? In other words, why do we need it in this test case?